pkg/webui/console/views/device-add/repository/device-selection/index.js

```diff
@@ -0,0 +1,115 @@
+// Copyright © 2021 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import classnames from 'classnames'
+
+import PropTypes from '@ttn-lw/lib/prop-types'
+
+import style from '../repository.styl'
+import { useRepositoryContext } from '../context'
+import { isOtherOption } from '../../utils'
+import {
+  selectBrand,
+  selectModel,
+  selectHwVersion,
+  selectFwVersion,
+  hasSelectedBrand,
+  hasSelectedModel,
+  hasSelectedHwVersion,
+  hasSelectedFwVersion,
+} from '../reducer'
+
+import BrandSelect from './brand-select'
+import ModelSelect from './model-select'
+import HardwareVersionSelect from './hw-version-select'
+import FirmwareVersionSelect from './fw-version-select'
+import BandSelect from './band-select'
+
+const Selection = props => {
+  const { onBrandChange, onModelChange, onHwVersionChange, onFwVersionChange, onBandChange } = props
+  const state = useRepositoryContext()
+
+  const brand = selectBrand(state)
+  const model = selectModel(state)
+  const hardwareVersion = selectHwVersion(state)
+  const firmwareVersion = selectFwVersion(state)
+
+  const hasBrand = hasSelectedBrand(state) && !isOtherOption(brand)
+  const hasModel = hasSelectedModel(state) && !isOtherOption(model)
+  const hasHwVersion = hasSelectedHwVersion(state) && !isOtherOption(hardwareVersion)
+  const hasFwVersion = hasSelectedFwVersion(state) && !isOtherOption(firmwareVersion)
+
+  return (
+    <div className={style.configurationSection}>
+      <BrandSelect
+        className={classnames(style.select, style.selectS)}
+        name="version_ids.brand_id"
+        required
+        onChange={onBrandChange}
+      />
+      {hasBrand && (
+        <ModelSelect
+          className={classnames(style.select, style.selectS)}
```

Comment:
```suggestion
          inputWidth="s"
```

Please check whether you can achieve same results with the built in sizes.

```
PropTypes.inputWidth = PropTypes.oneOf(['xxs', 'xs', 's', 'm', 'l', 'full'])
```

Reply:
Unfortunately setting `inputWidth` on inputs doesnt work here. I decided to set width here so I dont have do spend time on adjusting field styles and possibly breaking styling in other forms. However, I do agree that `inputWidth` should be used in this case.